### PR TITLE
Add subnavigation component

### PR DESCRIPTION
**Pull request by Xb0X**
Add subnavigation component (module)

### CODE REVIEW

File: src/modules/SubNavigation/index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+SubNavigation.defaultProps = {
+  items: [],
+};
+
+SubNavigation.propTypes = {
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number,
+      label: PropTypes.string,
```
Comment (bdo) [inline src/modules/SubNavigation/index.js]: label should be required

File: src/modules/SubNavigation/index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+);
+
+SubNavigation.defaultProps = {
+  items: [],
+};
+
+SubNavigation.propTypes = {
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number,
```
Comment (rogerfranklin2) [inline src/modules/SubNavigation/index.js]: id is not required but will give a console error if not provided

File: src/modules/SubNavigation/index.js (hunk 1 above)
Comment (rogerfranklin2) [inline src/modules/SubNavigation/index.js]: Again not required? If there is no label there probably isn't any point in having a link

File: src/modules/SubNavigation/index.stories.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { object } from '@storybook/addon-knobs';
+import { withInfo } from '@storybook/addon-info';
+import SubNavigation from './index';
+
+const subNavigationStories = storiesOf('SubNavigation', module);
+const items = [
+  {
+    id: 1,
```
Comment (bdo) [inline src/modules/SubNavigation/index.stories.js]: remove all ids

File: src/modules/SubNavigation/index.test.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import SubNavigation from '.';
+
+const items = [
+  {
+    id: 1,
```
Comment (bdo) [inline src/modules/SubNavigation/index.test.js]: remove all ids

File: src/modules/SubNavigation/index.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+  :hover {
+    color: white;
+  }
+`;
+
+const SubNavigation = ({ items, ...props }) => (
+  <StyledWrapper {...props}>
+    <StyledItems>
+      {items.map(item => (
```
Comment (bdo) [inline src/modules/SubNavigation/index.js]: Use destructuring:
```javascript
  {items.map(({ label, linkProps: { href, ...otherLinkProps } })=> (
```

File: src/modules/SubNavigation/index.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        <StyledItem key={item.id}>
+          <StyledLink {...item.linkProps}>{item.label}</StyledLink>
```
Comment (bdo) [inline src/modules/SubNavigation/index.js]: Adding the `href` explicitly has the advantage that the reader sees where it's used (it's referenced in `PropTypes` below).
```javascript
  <StyledLink href={href} {...otherLinkProps}>{label}</StyledLink>
```

File: src/modules/SubNavigation/index.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    </StyledItems>
+  </StyledWrapper>
+);
+
+SubNavigation.defaultProps = {
+  items: [],
+};
+
+SubNavigation.propTypes = {
+  items: PropTypes.arrayOf(
```
Comment (bdo) [inline src/modules/SubNavigation/index.js]: `items` should be required as it probably does not make sense for a submenu to have no items?

File: src/modules/SubNavigation/index.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      {items.map(item => (
+        <StyledItem key={item.id}>
+          <StyledLink {...item.linkProps}>{item.label}</StyledLink>
+        </StyledItem>
+      ))}
+    </StyledItems>
+  </StyledWrapper>
+);
+
+SubNavigation.defaultProps = {
```
Comment (bdo) [inline src/modules/SubNavigation/index.js]: Remove `defaultProps` if you agree with my next comment 😉

**Pull request merged**